scanners/nessus/nessus_none.py

```diff
@@ -0,0 +1,136 @@
+        # XXX self.config is already a dict with raw config values
+        self.cfg = dacite.from_dict(data_class=NessusConfig, data=nessus_config_section)
+        self.sleep_interval: int = 20
+        self._connect()
+
+    def _connect(self):
+        logging.debug(f"Connecting to nessus instance at {self.cfg.server.url}")
+        try:
+            self._nessus_client = PyNessusPro(
+                self.cfg.server.url,
+                self.cfg.server.username,
+                self.cfg.server.password,
+                log_level="debug",
+            )
+        except requests.exceptions.RequestException as e:
+            logging.error(f"Failed to connect to {self.cfg.server.url}: {e}")
+            raise
+
+    @property
+    def nessus_client(self) -> PyNessusPro:
+        if self._nessus_client is None:
+            raise RuntimeError(f"Nessus client not connected: {self.state}")
+        return self._nessus_client
+
+    @property
+    def scan_id(self) -> int:
+        if self._scan_id is None:
+            raise RuntimeError("scan_id is None")
+        return self._scan_id
+
+    def setup(self):
+        logging.debug(f"Creating new scan named {self.cfg.scan.folder}/{self.cfg.scan.name}")
+        self._scan_id = self.nessus_client.new_scan(
+            name=self.cfg.scan.name,
+            targets=self.cfg.scan.targets_as_str(),
+            folder=self.cfg.scan.folder,
+            create_folder=True,
+        )
+
+        # only user-created scan policies seem to be identified and must be
+        # created with the name used in the config as a prerequisite
+        if self.cfg.scan.policy:
+            logging.debug(f"Setting scan policy to {self.cfg.scan.policy}")
+            self.nessus_client.set_scan_policy(scan_id=self.scan_id, policy=self.cfg.scan.policy)
+
+        self.state = State.READY
+
+    def run(self):
+        if self.state != State.READY:
+            raise RuntimeError(f"[nessus] unexpected state: READY != {self.state}")
+        # State that we want the scan to launch immediately
+        logging.debug("Launching scan")
+        self.nessus_client.set_scan_launch_now(scan_id=self.scan_id, launch_now=True)
+
+        # Tell nessus to create and launch the scan
+        self.nessus_client.post_scan(scan_id=self.scan_id)
+
+        # Wait for the scan to complete
+        while self.nessus_client.get_scan_status(self.scan_id)["status"] not in END_STATUSES:
+            time.sleep(self.sleep_interval)
+            logging.debug("Waiting {self.sleep_interval}s for scan to finish")
+            logging.info(self.nessus_client.get_scan_status(self.scan_id))
+
+    def postprocess(self):
+        # After scan is complete, download report in csv, nessus, and html format
+        # Path and any folders must already exist in this implementation
+        logging.debug("Retrieving scan reports")
+        scan_reports = self.nessus_client.get_scan_reports(self.scan_id, self.results_dir)
+        print(scan_reports)
```

Comment:
We may want to use logging instead of print() for consistency?

Reply:
switched to logging.debug